pkg/controllers/clusterexternalsecret/util.go

```diff
@@ -22,17 +22,31 @@ import (
 )
 
 func NewClusterExternalSecretCondition(failedNamespaces map[string]error, namespaceList *v1.NamespaceList) *esv1beta1.ClusterExternalSecretStatusCondition {
-	conditionType := getConditionType(failedNamespaces, namespaceList)
-	condition := &esv1beta1.ClusterExternalSecretStatusCondition{
-		Type:   conditionType,
-		Status: v1.ConditionTrue,
+	switch {
+	case len(namespaceList.Items) == 0:
+		return &esv1beta1.ClusterExternalSecretStatusCondition{
+			Type:    esv1beta1.ClusterExternalSecretNotReady,
+			Status:  v1.ConditionTrue,
+			Message: errNamespaceNotFound,
+		}
+	case len(failedNamespaces) == 0:
+		return &esv1beta1.ClusterExternalSecretStatusCondition{
+			Type:   esv1beta1.ClusterExternalSecretReady,
+			Status: v1.ConditionTrue,
+		}
+	case len(failedNamespaces) < len(namespaceList.Items):
+		return &esv1beta1.ClusterExternalSecretStatusCondition{
+			Type:    esv1beta1.ClusterExternalSecretPartiallyReady,
+			Status:  v1.ConditionTrue,
+			Message: errNamespacesFailed,
+		}
```

Comment:
```suggestion
	switch {
	case len(namespaceList.Items) == 0:
		return &esv1beta1.ClusterExternalSecretStatusCondition{
			Type:    esv1beta1.ClusterExternalSecretReady,
			Status:  v1.ConditionFalse,
			Message: errNamespaceNotFound,
		}
	case len(failedNamespaces) == 0:
		return &esv1beta1.ClusterExternalSecretStatusCondition{
			Type:   esv1beta1.ClusterExternalSecretReady,
			Status: v1.ConditionTrue,
		}
	case len(failedNamespaces) < len(namespaceList.Items):
		return &esv1beta1.ClusterExternalSecretStatusCondition{
			Type:    esv1beta1.ClusterExternalSecretPartiallyReady,
			Status:  v1.ConditionTrue,
			Message: errNamespacesFailed,
		}
```

I find it superfluous that we have both a `Ready` and `NotReady` condition. 

With a bit fumbling around i ended up in a status of:
1. deploy ces with a matching selector
2. change selector to something that doesn't match
```yaml
status:
  conditions:
  - status: "True"
    type: Ready
  - message: no namespace matches
    status: "True"
    type: NotReady
```

Which doesn't make sense. I'd remove the `NotReady` condition type and instead work with the `Ready` condition and set it to false. That way we get a consistent status. WDYT?

Reply:
Good catch, thank you! 🙇 
I agree that the current design is a little too complicated, which makes it error-prone. Let me fix that!

Reply:
Should we remove `PartiallyReady` as well? It seems if we substitute `NotReady & true` with `Ready & false`, `PartiallyReady & true` is not mutually exclusive with `Ready& false` anymore. So I personally think it makes more sense if we just remove `PartiallyReady` and use `Ready & false` for that case as well 🤔 

I've fixed the PR that way, please let me know what you think about that 🙇

Reply:
Agree, that makes sense! I'll do another round of manual testing in the next days.